.github/workflows/sonar.yml

```diff
@@ -0,0 +1,21 @@
+name: Build
+on:
+  push:
+    branches:
+      - master
+      - dev
+  pull_request:
+    types: [opened, synchronize, reopened]
+jobs:
+  sonarcloud:
+    name: SonarCloud
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0  # Shallow clones should be disabled for a better relevancy of analysis
+      - name: SonarCloud Scan
+        uses: SonarSource/sonarcloud-github-action@master
```

Comment:
Does this mean that it only scans master?

Reply:
No, this means it's using a job written by SonarCloud, hosted on the master branch of their actions repo.

You can see their repo [here.](https://github.com/SonarSource/sonarcloud-github-action)

Reply:
So this should work for any branch correct? or just the protected branches(master,dev)?

Reply:
The scan runs on whatever branch triggered the action, in this case since it was a PR it runs on jc/sonar-action, but when dev triggers a scan because of a push to it, the scan will run on dev. Same goes for master. We can add branches to be scanned on every push under the `branches:` line on line 4, right now it's just dev and master.

Reply:
Basically, the way it's set up now it will run a scan whenever a PR is opened or updated (line 8), and whenever master or dev has commits pushed to it.

Reply:
OK, so now it will run only on merge to dev or master correct?

Reply:
Yes, but also on PR branches when a PR is opened or updated.

Reply:
Cool, merging